### PR TITLE
Apply code quality fixes and doc

### LINKED ISSUE #297: Deterministic bootstrapping
Iterator over map in bootstrap.go `setAllPoEContractsInstanceMigrators`
https://github.com/confio/tgrade/security/code-scanning/11?query=ref%3Arefs%2Fpull%2F288%2Fmerge+ref%3Arefs%2Fpull%2F288%2Fhead+ref%3Arefs%2Fheads%2F61-genesis_validation

### LINKED ISSUE #317: Check/address all  code scanning warnings
Tgrade: Dry Run Net
Check/address all  code scanning warnings: https://github.com/CosmWasm/wasmd/security/code-scanning (about 20 errors)

time box: 1 day (S)
outcome: a quick fix or follow up tickets

### PR DESCRIPTION
**Pull request by alpe**
Resolves #317
Resolves #297

### CODE REVIEW

File: x/poe/bootstrap.go [hunk 1]
```diff
@@ -344,21 +344,18 @@ func addToTrustedCircle(ctx sdk.Context, contractAddr sdk.AccAddress, tk twasmKe
...
-			continue
-		}
-		addr, err := poeKeeper.GetPoEContractAddress(ctx, contractType)
+	var rspErr error
+	types.IteratePoEContractTypes(func(tp types.PoEContractType) bool {
+		addr, err := poeKeeper.GetPoEContractAddress(ctx, tp)
 		if err != nil {
-			return sdkerrors.Wrapf(err, "failed to find contract address for %s", name)
+			rspErr = sdkerrors.Wrapf(err, "failed to find contract address for %s", tp.String())
 		}
```
Comment (pinosu) [inline x/poe/bootstrap.go:352]: here it is missing the return  statement. rspErr will be overwritten at line 354

File: x/poe/keeper/keeper.go [hunk 2]
```diff
@@ -47,17 +47,16 @@ func NewKeeper(
...
+// GetPoEContractAddress get the stored contract address for the given type
 func (k Keeper) GetPoEContractAddress(ctx sdk.Context, ctype types.PoEContractType) (sdk.AccAddress, error) {
-	if ctype == types.PoEContractTypeUndefined {
-		return nil, sdkerrors.Wrap(wasmtypes.ErrInvalid, "contract type")
-	}
-	if _, ok := types.PoEContractType_name[int32(ctype)]; !ok {
-		return nil, sdkerrors.Wrap(wasmtypes.ErrNotFound, "contract type")
+	if err := ctype.ValidateBasic(); err != nil {
+		return nil, sdkerrors.Wrap(err, "contract type")
 	}
```
Comment (pinosu) [inline x/poe/keeper/keeper.go:60]: nice!

File: x/poe/keeper/keeper.go [hunk 3]
```diff
@@ -67,14 +66,16 @@ func (k Keeper) GetPoEContractAddress(ctx sdk.Context, ctype types.PoEContractTy
...
 func (k Keeper) IteratePoEContracts(ctx sdk.Context, cb func(types.PoEContractType, sdk.AccAddress) bool) {
 	prefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), types.ContractPrefix)
 	iter := prefixStore.Iterator(nil, nil)
 	defer iter.Close()
 	for ; iter.Valid(); iter.Next() {
 		// cb returns true to stop early
-		ctype := types.PoEContractType_value[string(iter.Key())]
-		if cb(types.PoEContractType(ctype), iter.Value()) {
+		ctype := types.PoEContractTypeFrom(string(iter.Key()))
+		if cb(ctype, iter.Value()) {
```
Comment (pinosu) [inline x/poe/keeper/keeper.go:78]: 👍

File: x/poe/types/poe.go [hunk 4]
```diff
@@ -14,10 +16,30 @@ func (t PoEContractType) ValidateBasic() error {
...
+func IteratePoEContractTypes(cb func(tp PoEContractType) bool) {
+	names := make([]string, 0, len(PoEContractType_name)-1)
+	for _, v := range PoEContractType_name {
+		if v == PoEContractTypeUndefined.String() {
+			continue
+		}
+		names = append(names, v)
+	}
+	sort.Strings(names)
+	for _, v := range names {
```
Comment (pinosu) [inline x/poe/types/poe.go:40]: 👍

**Review by pinosu — CHANGES_REQUESTED** [review]
looks great! 
Only one minor comment to fix (missing return statement).

File: x/poe/bootstrap.go (hunk 1 above)
Comment (alpe) [reply x/poe/bootstrap.go:352]: 👍  I should abort early. Good 👁️

**Review by pinosu — APPROVED** [review]
lgtm 👍

**Pull request merged**